packages/expo-updates/src/Updates.types.ts

```diff
@@ -7,7 +7,7 @@ export enum UpdateEventType {
 }
 
 // TODO(eric): move source of truth for manifest type to this module
-export type Manifest = typeof Constants.manifest;
+export type Manifest = typeof Constants.manifest | typeof Constants.manifest2;
```

Comment:
What do you think of something like this?
```suggestion
export type ClassicManifest = typeof Constants.manifest;
export type Manifest = ClassicManifest | typeof Constants.manifest2;
// modern manifest type is intentionally not exported, since the plan is to call it just "Manifest" in the future
```

and then later just:
```
export type Manifest = typeof Constants.manifest2;
export type ClassicManifest = typeof Constants.manifest;
```